<launch>
	<node name="hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
		<remap from="scan" to="tilt_scan" />
    	<param name="min_ang" type="double" value="-1.57"/>
		<param name="max_ang" type="double" value="1.57"/>
   		<param name="port" type="string" value="/dev/ttyACM1"/>
   		<param name="frame_id" type="string" value="/tilt_laser_link"/>
   		<param name="intensity" type="bool" value="false"/>
 	</node>
 	<!-- Filter for tilt laser shadowing/veiling -->
	<!--node pkg="laser_filters" type="scan_to_cloud_filter_chain" respawn="true" name="tilt_shadow_filter">
  		<remap from="scan" to="tilt_scan" />
  		<remap from="cloud_filtered" to="tilt_cloud_filter" />
  		<param name="target_frame" value="base_link" />
  		<param name="high_fidelity" value="true" />
		<rosparam command="load" file="$(find robot_connect)/launch/config/tilt_laser_filters.yaml" />
	</node-->

	<node name="tilt_laser_preprocess" pkg="robot_connect" type="tilt_laser_preprocess"/>
	
 	<node pkg="semantic_point_annotator" type="sac_inc_ground_removal_node" name="sac_ground_removal" output="screen">
  	<remap from="tilt_laser_cloud_filtered" to="tilt_scan_cloud" />
  	<remap from="cloud_ground_filtered" to="ground_object_cloud" />
  	<param name="z_threshold" value="0.30" />
  	<param name="sac_min_points_per_model" value="40" />
  	<param name="sac_distance_threshold" value="0.10" />
  	<param name="planar_refine" value="0" />
  	<param name="fixed_frame" value="odom" />
  	<param name="laser_tilt_mount_frame" value="tilt_laser_link" />
  	<param name="robot_footprint_frame" value="base_link" />
  </node>
  
  <!--node name="prismatic_control" pkg="robot_connect" type="prismatic_control" output="screen"/-->
</launch>
